packages/db/src/search/ExecutionSearch.ts

```diff
@@ -0,0 +1,152 @@
+    } else {
+      searchQuery.leftJoin(collectionsTable, `${this.tableName}.collection_cumulus_id`, `${collectionsTable}.cumulus_id`);
+    }
+
+    if (this.searchAsync()) {
+      countQuery.innerJoin(asyncOperationsTable, `${this.tableName}.async_operation_cumulus_id`, `${asyncOperationsTable}.cumulus_id`);
+      searchQuery.innerJoin(asyncOperationsTable, `${this.tableName}.async_operation_cumulus_id`, `${asyncOperationsTable}.cumulus_id`);
+    } else {
+      searchQuery.leftJoin(asyncOperationsTable, `${this.tableName}.async_operation_cumulus_id`, `${asyncOperationsTable}.cumulus_id`);
+    }
+
+    if (this.searchParent()) {
+      countQuery.innerJoin(`${this.tableName} as ${this.tableName}_parent`, `${this.tableName}.parent_cumulus_id`, `${this.tableName}_parent.cumulus_id`);
+      searchQuery.innerJoin(`${this.tableName} as ${this.tableName}_parent`, `${this.tableName}.parent_cumulus_id`, `${this.tableName}_parent.cumulus_id`);
+    } else {
+      searchQuery.leftJoin(`${this.tableName} as ${this.tableName}_parent`, `${this.tableName}.parent_cumulus_id`, `${this.tableName}_parent.cumulus_id`);
+    }
+    return { countQuery, searchQuery };
+  }
+
+  /**
+   * Build queries for infix and prefix
+   *
+   * @param params
+   * @param params.countQuery - query builder for getting count
+   * @param params.searchQuery - query builder for search
+   * @param [params.dbQueryParameters] - db query parameters
+   */
+  protected buildInfixPrefixQuery(params: {
+    countQuery: Knex.QueryBuilder,
+    searchQuery: Knex.QueryBuilder,
+    dbQueryParameters?: DbQueryParameters,
+  }) {
+    const { countQuery, searchQuery, dbQueryParameters } = params;
+    const { infix, prefix } = dbQueryParameters ?? this.dbQueryParameters;
+    if (infix) {
+      countQuery.whereLike(`${this.tableName}.arn`, `%${infix}%`);
+      searchQuery.whereLike(`${this.tableName}.arn`, `%${infix}%`);
+    }
+    if (prefix) {
+      countQuery.whereLike(`${this.tableName}.arn`, `${prefix}%`);
+      searchQuery.whereLike(`${this.tableName}.arn`, `${prefix}%`);
+    }
+  }
+
+  /**
+   * Translate postgres records to api records
+   *
+   * @param pgRecords - postgres records returned from query
+   * @returns translated api records
+   */
+  protected translatePostgresRecordsToApiRecords(pgRecords: ExecutionRecord[])
+    : Partial<ApiExecutionRecord[]> {
+    log.debug(`translatePostgresRecordsToApiRecords number of records ${pgRecords.length} `);
+    const results: ApiExecutionRecord[] = [];
+    pgRecords.map((executionRecord) => {
+      const { collectionName, collectionVersion, asyncOperationId, parentArn } = executionRecord;
+      const collectionId = collectionName && collectionVersion ?
+        constructCollectionId(collectionName, collectionVersion) : undefined;
+      const result = translatePostgresExecutionToApiExecutionWithoutDbQuery({
+        executionRecord,
+        collectionId,
+        asyncOperationId,
+        parentArn,
+      });
```

Comment:
Can you refer to GranuleSearch?  It should support `this.dbQueryParameters.fields`